File: NicolasNuñez/Python/clase5/ejercicio2.py
'''
Ejercicio 2: Factorial de un numero positivo.
Hacer un programa para calcular el factorial de un numero positivo
EJ: 5! = 5x4x3x2x1 = 120
'''

num = int((input("Ingrese un numero: ")))

factorial = 1

for i in range(num, 0, -1):
    factorial *= i

print(f"El factorial de {num} es {factorial}")

File: MarcosSenn/Java/Clases/src/Clase7/Ejercicio2.java
package Clase7;
import javax.swing.JOptionPane;
import java.util.Scanner;

public class Ejercicio2 {
    //factorial de un numero con JOptionPane
    public static void main(String[] args){
        String numero = JOptionPane.showInputDialog("Ingrese un numero para calcular su factorial");
        int num = Integer.parseInt(numero);
       System.out.println( factorial(num));
    }
    public static int factorial(int num){
        if(num == 0){
            return 1;
        }
        if(num ==1){return 1;}
        return num * factorial(num - 1);
    }
}